### PR TITLE
cleanup-book: remove docs/book/src/migration/testdata

### PR DESCRIPTION
**Pull request by camilamacedo86**
Closes: #1279

The mainly complain of #1279 were the outdated mock testdata in the book which was just used as an example over the final results of the migration from v1 o v2. 

In this way, since the V1 is deprecated and no longer for too long then, shows quite safe to remove the old mock data that was just added as an illustration.

### CODE REVIEW

**Comment by camilamacedo86** [comment]
Hi @Adirio @mengqiy, 

Could you please help by checking this one? Wdyt? Are you ok with?

File: testdata/project-v3-addon/go.mod [hunk 1]
```diff
@@ -6,8 +6,10 @@ require (
...
-	k8s.io/apimachinery v0.17.2
-	k8s.io/client-go v0.17.2
-	sigs.k8s.io/controller-runtime v0.5.0
-	sigs.k8s.io/kubebuilder-declarative-pattern v0.0.0-20200415210853-85eb326a6add
+	gonum.org/v1/netlib v0.0.0-20190331212654-76723241ea4e // indirect
+	k8s.io/apimachinery v0.18.2
+	k8s.io/client-go v0.18.2
+	sigs.k8s.io/controller-runtime v0.6.0
+	sigs.k8s.io/kubebuilder-declarative-pattern v0.0.0-20200507203943-d901af431e3a
+	sigs.k8s.io/structured-merge-diff v1.0.1-0.20191108220359-b1b620dd3f06 // indirect
```
Comment (camilamacedo86) [inline testdata/project-v3-addon/go.mod:14]: Indirect deps which were updated with sigs.k8s.io/kubebuilder-declarative-pattern

**Comment by Adirio** [comment]
/hold

See https://github.com/kubernetes-sigs/kubebuilder/issues/1279#issuecomment-627831376

TLDR: Now that the purpose of the testdata folder is clear, IMHO we should not remove it.

**Comment by camilamacedo86** [comment]
HI @mengqiy @droot, 

IHMO it is quite safe we remove the V1 example from the books since:
- V1 code is deprecated, no longer support and removed
- Users are still able to follow up on the steps. The example do not add too much value at all.
- Also, since it uses dep and it is no longer a required at all the following error will be faced `

> Cannot run program "/Users/camilamacedo/go/bin/dep" (in directory "/Users/camilamacedo/go/src/sigs.k8s.io/kubebuilder/docs/book/src/migration/testdata/gopath/project-v1"): error=2, No such file or directory

WDYT?

**Comment by camilamacedo86** [comment]
hi @pwittrock,

I saw that you have been helping with the reviews. really tks btw. 🥇 So, this one is a case that is ok to be merged but requires more opinions to reach a consensus. if you have time would be great get your input.

**Comment by camilamacedo86** [comment]
Hi, @gabbifish @pwittrock @DirectXMan12 it is open for a while. WDYT? Are you ok with we removing the mock data of v1 since it is no longer supported a while?

**Comment by camilamacedo86** [comment]
I am removing the 
/hold cancel

Otherwise will be hard to get reviews.
